RN35/yacp
# LAB-367: Modernize the IDE page

# Proposed Changes

- Migrated from create-react-app to Next.js 
- Added Material UI support 
- Removed the backend API code 
   - Next.js has capability to write APIs (https://nextjs.org/docs/api-routes/introduction)
- Removed the express.js based backend code execution code
   - This will be offloaded to Judge0 in a future JIRA (https://github.com/judge0/judge0)
- Created a basic layout ( NavBar, Drawer, Profile Menu) - All pages will inherit this (see: https://nextjs.org/docs/basic-features/layouts)
- Created the IDE page using Material UI components
   - Tabs to show (Problem description, Solution and past submissions) - These contain placeholder content for now
   - Integrated a code editor - Monaco (https://github.com/microsoft/monaco-editor)

# Jira Stories

- [LAB-367](https://oregonstate-innovationlab.atlassian.net/browse/LAB-367)

# Checklist

- [x] This has been tested
- [x] Self-documenting code
- [x] Code is linted

# Instructions to review
## How to Run
- Clone the repository: `git clone git@github.com:RN35/yacp.git`
- Change dir: `cd yacp`
- Checkout branch migrate-to-nextjs: `git checkout migrate-to-next-js`
- Install dependencies: `npm install`
- Spin up development server: `npm run dev`

## How to Validate
- Open `http://localhost:3000/problems` in any web browser
- This is just the UI, so just make sure everything renders correctly and there are no errors in the console 


# Screenshots 

### Drawer opens on the clicking the `hamburger menu`

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/15828512/221331115-b828b355-7799-482d-b605-82c1d9793f7e.png">

### A dropdown menu opens on clicking the profile icon on the top-right of the navbar

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/15828512/221331134-4528da74-a10a-40e5-ba36-8056ec4f6d7c.png">


### The IDE page

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/15828512/221331101-e65f373f-c427-41e5-8c49-9130cd5d270f.png">
